### PR TITLE
RobotHardwarsStats Changes

### PR DESCRIPTION
**Pull request by Nummun14**

### CODE REVIEW

**Review by Strflightmight09 — APPROVED** [review]

File: src/main/java/org/trigon/hardware/RobotHardwareStats.java [hunk 1]
```diff
@@ -5,9 +5,9 @@ public class RobotHardwareStats {
     private static boolean IS_REPLAY = false;
     private static double PERIODIC_TIME_SECONDS = 0.02;
 
-    public static void setCurrentRobotStats(boolean isReal, boolean isSimulation, boolean isReplay) {
-        IS_SIMULATION = isSimulation && !isReal;
-        IS_REPLAY = isReplay && !isReal;
+    public static void setCurrentRobotStats(boolean isReal, ReplayType replayType) {
+        IS_SIMULATION = isReal || replayType.equals(ReplayType.NONE) ? !isReal : replayType.equals(ReplayType.SIMULATION_REPLAY);
+        IS_REPLAY = !isReal && !replayType.equals(ReplayType.NONE);
```
Comment (levyishai) [inline src/main/java/org/trigon/hardware/RobotHardwareStats.java]: Make the code logic less confusing with an if. Some of the logic here also seems flawed (there doesn't seem to be a diff between SIMULATION_REPLAY and REAL_REPLAY, etc.)

**Review by levyishai — CHANGES_REQUESTED** [review]

File: src/main/java/org/trigon/hardware/RobotHardwareStats.java (hunk 1 above)
Comment (Nummun14) [reply src/main/java/org/trigon/hardware/RobotHardwareStats.java]: The difference between SIMULATION_REPLAY and REAL_REPLAY is that in sim reply IS_SIMULATION will be true , that's also what the code does

Comment (levyishai) [reply src/main/java/org/trigon/hardware/RobotHardwareStats.java]: That's correct, my bad. 
Still, I think the usage of the ternary operator here is a bit confusing.

File: src/main/java/org/trigon/hardware/RobotHardwareStats.java [hunk 2]
```diff
@@ -5,9 +5,14 @@ public class RobotHardwareStats {
     private static boolean IS_REPLAY = false;
     private static double PERIODIC_TIME_SECONDS = 0.02;
 
-    public static void setCurrentRobotStats(boolean isReal, boolean isSimulation, boolean isReplay) {
-        IS_SIMULATION = isSimulation && !isReal;
-        IS_REPLAY = isReplay && !isReal;
+    public static void setCurrentRobotStats(boolean isReal, ReplayType replayType) {
+        if (isReal && replayType.equals(ReplayType.NONE)) {
```
Comment (levyishai) [inline src/main/java/org/trigon/hardware/RobotHardwareStats.java]: And?

File: src/main/java/org/trigon/hardware/RobotHardwareStats.java [hunk 3]
```diff
@@ -5,9 +5,14 @@ public class RobotHardwareStats {
...
+            IS_SIMULATION = !isReal;
+            IS_REPLAY = false;
+            return;
+        }
+        IS_SIMULATION = replayType.equals(ReplayType.SIMULATION_REPLAY);
```
Comment (levyishai) [inline src/main/java/org/trigon/hardware/RobotHardwareStats.java:15]: Maybe add a whitespace here?

**Review by levyishai — CHANGES_REQUESTED** [review]

**Review by Strflightmight09 — APPROVED** [review]

**Review by levyishai — APPROVED** [review]
Implement in the template.

**Pull request merged**